### DISCUSSION TITLE
Audio interface name randomly changing under Linux

### DISCUSSION THREAD

**Discussion by newVideophile** [GrandOrgue/grandorgue#1815] [Help]
This is probably not a GO issue, but in case someone has already faced - and solved - this issue, I am asking here for help. 
I use GO on a Linux Mint laptop with a Motu M4 audio interface. The problem is that the latter is not always detected with the same name. For example:
- Boot 1 = PortAudio: ALSA: M4: USB Audio (hw:3,0)
- Boot 2 = no PortAudio interface any more; it is only seen as: RtAudio: alsa: hw:M4,0
- Boot 3 = PortAudio: ALSA: M4: USB Audio (hw:5,0)

Is there a way for force Linux to always detect my audio interface with the same name so that I don't need to constantly adjust the GO settings (or close and restart it)?

**Comment by oleg68** [comment]
See #1265

**Reply by newVideophile** [reply]
Thank you oleg. Before posting, I searched the forum, but I didn't manage to find this post. 

So the problem is known, but unresolved so far. Sad news :-(

**Comment by davidgritter** [comment]
If you only have one M4 audio usb device, then you can use the jack audio connection kit to always connect to the same device.  If you want to automatically start Grandorgue in linux, you would create a startup shell script  similar to the below

jackd -R -P85 -dalsa -dhw:M4 -r48000 -p256 -n4 -P -o8  &>~/jack.log &
sleep 3
GrandOrgue &>~/grandorgue.log &
sleep 3
qjackctl &

where  the -dhw:M4 specifies the alsa device name 'M4'  you could also use, for example  -dhw:M4,0, if you want to specify a subdevice

Then configure GrandOrgue ins settings, Audio to use RtAudio:jack:system  for the audio outputs

**Reply by newVideophile** [reply]
Many thanks for this, I will try.

Just a question, what is the difference between RtAudio and PortAudio? I prefer the latter, because it gives twice as less latency for the same buffer depth, according to both GO and my ears.

Besides, in some cases the Motu M4 interface will show up as RtAudio but not PortAudio. I'm afraid that in this case, the jack trick will not solve my problem.

**Reply by davidgritter** [reply]
Both RtAudio and Port Audio appear to be interfaces between GrandOrgue and the Available sound devices.   With Jack you don't need to use either. If you want to minimize latency; just select jack directly. This should minimize all latency.  (But if you select jack output directly in GrandOrgue, you will get standard jack outputs that you need to manually connect to the system inputs that Jack provides by default.  Using either PortAudio, or RtAudio makes these connections for you. There are tools in qjackctl to automate making these connections as well.)  I've always use RtAudio, but you could use PortAudio to make the connection to jack.

In either case, neither RtAudio nor Port audio needs to know anything about your sound card;  only jack needs to be able to connect to it.
